pkg/runtime/reconciler.go

```diff
@@ -194,10 +194,20 @@ func (r *resourceReconciler) reconcile(
 	res acktypes.AWSResource,
 ) (acktypes.AWSResource, error) {
 	if res.IsBeingDeleted() {
-		// Resolve references before deleting the resource.
-		// Ignore any errors while resolving the references
-		res, _ = rm.ResolveReferences(ctx, r.apiReader, res)
-		return r.deleteResource(ctx, rm, res)
+		// Determine whether we should retain or delete the resource
+		if r.getDeletionPolicy(res) == ackv1alpha1.DeletionPolicyDelete {
```

Comment:
Didn't know this was possible, I always thought it's not possible to use `==` for "similar" type aliases

Reply:
`DeletionPolicyDelete` is a string enum, so this is just a string comparison operation